scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/TypedParquet.scala

```diff
@@ -66,37 +49,26 @@ object TypedParquetSink {
   /**
    * Create typed parquet sink.
    * Here is an example:
-   *
-   *  case class SampleClassB(string: String, int: Int, double: Option[Double], a: SampleClassA)
-   *
-   *  class WriteSupport extends ParquetWriteSupport[SampleClassB] {
-   *    import com.twitter.scalding.parquet.tuple.macros.Macros._
-   *
-   *    override def writeRecord(r: SampleClassB, rc: RecordConsumer, schema: MessageType): Unit =
-   *      caseClassWriteSupport[SampleClassB](r, rc, schema)
-   *    override val rootSchema: String = caseClassParquetSchema[SampleClassB]
-   *  }
-   *
-   *  val sink = TypedParquetSink[SampleClassB, WriteSupport](Seq(outputPath))
+   *  import com.twitter.scalding.parquet.tuple.macros.Macros._
+   *  val sink = TypedParquetSink[SampleClass](Seq(outputPath),
+   *    caseClassParquetWriteSupport[SampleClass](caseClassParquetSchema[SampleClass]))
    *
    * @param paths paths of parquet I/O
-   * @param t Read support type tag
    * @tparam T Tuple type
-   * @tparam W Write support type
    * @return a typed parquet source.
    */
-  def apply[T, W <: ParquetWriteSupport[T]](paths: Seq[String])(implicit t: ClassTag[W]) =
-    new TypedFixedPathParquetTuple[T, ParquetReadSupport[T], W](paths, null, t.runtimeClass.asInstanceOf[Class[W]])
+  def apply[T](paths: Seq[String], writeSupport: ParquetWriteSupport[T]) =
+    new TypedFixedPathParquetTuple[T](paths, null, writeSupport)
 }
 
 /**
  * Typed Parquet tuple source/sink.
  */
-trait TypedParquet[T, R <: ParquetReadSupport[T], W <: ParquetWriteSupport[T]] extends FileSource with Mappable[T]
+trait TypedParquet[T] extends FileSource with Mappable[T]
   with TypedSink[T] with HasFilterPredicate {
 
-  val readSupport: Class[R]
-  val writeSupport: Class[W]
+  val readSupport: ParquetReadSupport[T]
```

Comment:
generally in scala it is better to make `def` abstract. making an abstract `val` can cause some issues (val init order bug).